src/WinRT.Impl.Generator/Generation/ImplGenerator.cs

```diff
@@ -268,13 +268,40 @@ private static void EmitTypeForwards(ModuleDefinition inputModule, ModuleDefinit
                     Attributes = AsmResolver.PE.DotNet.Metadata.Tables.TypeAttributes.Forwarder
                 });
             }
+
+            TypeReference interfaceIIDsTypeReference = inputModule.CreateTypeReference("ABI"u8, "InterfaceIIDs"u8);
+            if (interfaceIIDsTypeReference.Resolve() is TypeDefinition interfaceIIDsTypeDefinition)
+            {
+                // Forward the 'ABI.InterfaceIIDs' type as well, as it's used for IID resolution at runtime
+                implModule.ExportedTypes.Add(new ExportedType(
+                    implementation: projectionAssembly.ImportWith(implModule.DefaultImporter),
+                    ns: interfaceIIDsTypeDefinition.Namespace,
+                    name: interfaceIIDsTypeDefinition.Name)
+                {
+                    Attributes = AsmResolver.PE.DotNet.Metadata.Tables.TypeAttributes.Forwarder
+                });
+            }
         }
         catch (Exception e)
         {
             throw WellKnownImplExceptions.EmitTypeForwards(e);
         }
     }
 
+    /// <summary>
+    /// There are a couple types in the Windows SDK projection which are not public and vtables are generated for.
+    /// This checks whether it is one of those by checking the namespace.
+    /// </summary>
+    /// <param name="exportedType">The type to check.</param>
+    /// <returns>True if the namespace is one of the forwarded internal types; otherwise, false.</returns>
+    private static bool IsForwardedInternalType(TypeDefinition exportedType)
```

Comment:
These should all be temporary, let's see once I move the stream stuff if we can remove this too.